module3/lessons/week3/CreatingRelatedResources.md

```diff
@@ -0,0 +1,349 @@
+}
+```
+
+</section>
+
+<section class='instructor-notes' markdown='1'>
+This is a good opportunity to ask students to navigate through the implementation while you drive!
+</section>
+
+Now that we have some tests created, we can use them to guide our implementation!
+
+One main difference between this `new` view and the one we created for our Movie form is the information that needs to be passed in to the action, and then on to the view:
+
+```c#
+// GET: /movies/:movieId/reviews/new
+[Route("Movies/{movieId:int}/reviews/new")]
+public IActionResult New(int movieId)
+{
+    var movie = _context.Movies
+        .Where(m => m.Id == movieId)
+        .Include(m => m.Reviews)
+        .First();
+
+    ViewData["MovieTitle"] = movie.Title;
+    ViewData["MovieId"] = movie.Id;
+    return View();
+}
+```
+
+### Test the Create Action
+
+<section class='call-to-action' markdown='1'>
+
+**In Small Groups**  
+Using the tests we created to verify the new movie create action, write the tests for the third user story above.
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```C#
+[Fact]
+public async Task Create_AddsReview_RedirectsToMovieReviewsIndex()
+{
+    var context = GetDbContext();
+    var client = _factory.CreateClient();
+
+    Movie spaceballs = new Movie { Genre = "Comedy", Title = "Spaceballs" };
+    context.Movies.Add(spaceballs);
+    context.SaveChanges();
+
+    var formData = new Dictionary<string, string>
+    {
+        { "Rating", "5" },
+        { "Content", "Better than Star Wars" }
+    };
+
+    var response = await client.PostAsync($"/movies/{spaceballs.Id}/reviews", new FormUrlEncodedContent(formData));
+    var html = await response.Content.ReadAsStringAsync();
+
+    Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+    Assert.Contains($"/Movies/{spaceballs.Id}/reviews", response.RequestMessage.RequestUri.ToString());
+    Assert.Contains("5: Better than Star Wars", html);
+    Assert.DoesNotContain("4: Good. But, when will then be now?", html);
+}
+```
+
+</section>
+
+<section class='call-to-action' markdown='1'>
+
+**In Small Groups**  
+Using the tests we created to verify the new movie create action, write the tests for the third user story above.
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```c#
+[HttpPost]
+[Route("/Movies/{movieId:int}/reviews")]
+public IActionResult Create(int movieId, Review review)
+{
+    var movie = _context.Movies
+        .Where(m => m.Id == movieId)
+        .Include(m => m.Reviews)
+        .First();
+    movie.Reviews.Add(review);
+    _context.SaveChanges();
+
+    return RedirectToAction("index", new { movieId = movie.Id });
+}
+```
+
+</section>
+
+## Checks for Understanding
+* Why do we need to include the movie id in our route for reviews?
+* What would happen if we did not `.Include()` the reviews for a movie in our Reviews Index action?
```

Comment:
I like this! I had a question about this earlier, after this check for understanding might be the perfect time to "Try it and see!"